test/e2e/pd.go

```diff
@@ -130,16 +141,27 @@ var _ = Describe("PD", func() {
 		expectNoError(err, "Failed to create host1ROPod")
 
 		By("waiting up to 60 seconds for host0ROPod to start running")
-		expectNoError(waitForPodRunning(c, host0ROPod.Name, 60*time.Second), "host0ROPod not running after 60 seconds")
+		expectNoError(waitForPodRunning(c, host0ROPod.Name, 120*time.Second), "host0ROPod not running after 60 seconds")
 
 		By("waiting up to 60 seconds for host1ROPod to start running")
-		expectNoError(waitForPodRunning(c, host1ROPod.Name, 60*time.Second), "host1ROPod not running after 60 seconds")
+		expectNoError(waitForPodRunning(c, host1ROPod.Name, 120*time.Second), "host1ROPod not running after 60 seconds")
 
 		By("deleting host0ROPod")
 		expectNoError(podClient.Delete(host0ROPod.Name), "Failed to delete host0ROPod")
 
 		By("deleting host1ROPod")
 		expectNoError(podClient.Delete(host1ROPod.Name), "Failed to delete host1ROPod")
+
+		By(fmt.Sprintf("deleting PD %q", diskName))
+		err = deletePD(diskName, testContext.gceConfig.Zone)
+		for i := 0; i < 10; i++ {
+			Logf("Couldn't delete PD. Sleeping 2 seconds")
+			time.Sleep(2 * time.Second)
+			if err = deletePD(diskName, testContext.gceConfig.Zone); err == nil {
+				break
+			}
+		}
+		expectNoError(err, "Error deleting PD")
```

Comment:
If this becomes a common pattern, we could probably write a generic retry-on-error that just took a func. You're not mutating anything here, you're just pounding it. And I see why (stupid GCE case where we don't care, but really don't want to leak the resource).